orangecontrib/timeseries/timeseries.py

```diff
@@ -21,14 +21,35 @@ def __init__(self, *args, **kwargs):
         self._time_delta = None
         # Set default time variable to first TimeVariable
         try:
-            self.time_variable = next(var for var in self.domain.attributes + self.domain.metas
+            self.time_variable = next(var for var in self.domain.attributes
                                       if isinstance(var, TimeVariable))
         except (StopIteration, AttributeError):
             pass
 
     @classmethod
     def from_data_table(cls, table):
-        return table if isinstance(table, Timeseries) else Timeseries(table)
+        if isinstance(table, Timeseries):
+            return table
+
+        # Set default time variable to first TimeVariable in:
+        search = table.domain.attributes + table.domain.metas
+
+        # Is there a time variable we can use?
+        try:
+            time_variable = next(var for var in search
+                                 if var.is_time)
+            return Timeseries(table.domain, table, time_variable=time_variable)
```

Comment:
Probably a call should be Timeseries.from_table? 
`OrangeDeprecationWarning: Direct calls to Table's constructor are deprecated and will be removed. Replace this call with Table.from_table`

Reply:
Also, Timeseries doesn't have a time_variable parameter, hence tests fail (all widgets actually). You can add it to the constructor.

Reply:
> Probably a call should be Timeseries.from_table?
> `OrangeDeprecationWarning: Direct calls to Table's constructor are deprecated and will be removed. Replace this call with Table.from_table`

Replaced all of the constructor calls in timeseries.py, there's still some scattered around though.

> Also, Timeseries doesn't have a time_variable parameter, hence tests fail (all widgets actually). You can add it to the constructor.

😅 That was a remnant of an experiment